### PR TITLE
Improve the editor's Add Metadata dialog

### PR DESCRIPTION
**Pull request by Calinou**
Follow-up to https://github.com/godotengine/godot/pull/59452.

- Mention the node name in the dialog title.
- Improve error messages to be more descriptive.

## Preview

image

### CODE REVIEW

File: editor/editor_inspector.cpp [hunk 1]
```diff
@@ -3687,15 +3687,21 @@ void EditorInspector::_add_meta_confirm() {
...
-	if (name == "") {
-		error = TTR("Metadata can't be empty.");
-	} else if (!name.is_valid_identifier()) {
-		error = TTR("Invalid metadata identifier.");
-	} else if (object->has_meta(name)) {
-		error = TTR("Metadata already exists.");
+	if (p_name == "") {
+		error = TTR("Metadata name can't be empty.");
+	} else if (!p_name.is_valid_identifier()) {
+		error = TTR("Invalid metadata name (must be a valid identifier).");
```
Comment (LightningAA) [inline editor/editor_inspector.cpp:3696]:
```suggestion
		error = TTR("Metadata name must be a valid identifier.");
```

**Comment by YuriSizov** [comment]
Needs a review, but should be good to go otherwise.

**Review by akien-mga — APPROVED** [review]
Looks good to me.

**Comment by akien-mga** [comment]
Reminder to rebase, so we can remove this from the backlog of approved but unmerged PRs :)

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**